test/e2e/app-dir/no-react/no-react.test.ts

```diff
@@ -0,0 +1,20 @@
+import { nextTestSetup } from 'e2e-utils'
+
+describe('no-react', () => {
+  const { next } = nextTestSetup({
+    files: __dirname,
+    installCommand: 'pnpm remove react react-dom',
+  })
+
+  it('should not have react or react-dom in the package.json', async () => {
+    const packageJson = await next.readJSON('package.json')
+    expect(packageJson.dependencies).toHaveProperty('next')
+    expect(packageJson.dependencies).not.toHaveProperty('react')
```

Comment:
This isn't testing Next.js behavior. You're manually removing `react` and `react-dom` so obviously it's not in the package.json.

Let's do this by the books and add a commit testing the existing behavior and another commit making the change and updating that test so that we know what we actually changed.